cross-doc-explainer.md

```diff
@@ -0,0 +1,163 @@
+  // Don't capture navigation-bar animation when navigating to home
+  if (!new URL(event.destination.url).pathname.startsWith("/home"))
+    navigationBar.viewTransitionName = "none";
+});
+```
+
+### Capturing the new state
+
+The [new state is captured](https://drafts.csswg.org/css-view-transitions-1/#capture-new-state-algorithm) right before the first [rendering opportunity](https://html.spec.whatwg.org/#rendering-opportunity)
+of the new document. This allows the new document to use the
+[render-blocking mechanism](https://html.spec.whatwg.org/#render-blocking-mechanism) as a way to
+delay the transition.
+
+As shown in the chart above, that first rendering opportunity can come in two cases, either
+it's a newly initialized document that's no longer [render-blocked](https://html.spec.whatwg.org/multipage/dom.html#render-blocked), or it's a document that's been frozen due to back-forward cache
+or prerendered, and is now being activated.
+
+Note: relying on the [render-blocking](https://html.spec.whatwg.org/multipage/dom.html#render-blocked) mechanism is limited, as only a limited set of elements participate in that
+mechanism. See [proposal to extend it](whatwg/html#9332).
+
+## JavaScript API
+
+To fullfill the design principle of making cross-document transitions [compatible with same-document transitions](#compatible-with-same-document-transitions), cross-document transition need to be as
+customizable as same-document transitions, while allowing good defaults that work declaratively.
+
+Same document transitions can be programatically extended (until the [updateCallbackDone](https://drafts.csswg.org/css-view-transitions-1/#dom-viewtransition-updatecallbackdone) promise is fullfilled), [skipped](https://drafts.csswg.org/css-view-transitions-1/#dom-viewtransition-skiptransition), programatically animated using the [ready](https://drafts.csswg.org/css-view-transitions-1/#dom-viewtransition-ready) promise, and their [end time can be
+observed](https://drafts.csswg.org/css-view-transitions-1/#dom-viewtransition-finished).
+
+To accomplish the same control and observability, the developer would need access to a
+[ViewTransition](https://drafts.csswg.org/css-view-transitions-1/#the-domtransition-interface)
+object. There are several ways to approach this, and this is an open issue:
+
+1. Expose it via `document.pendingViewTransition`. This would be available only before the document
+gets render-unblocked for the first time, and right at reactivation.
+
+1. Fire a special `reveal` event at both lifecycle moments, with a `ViewTransition` object. See whatwg/html#9315, w3c/csswg-drafts#8682, and w3c/csswg-drafts#8805. Note that this event is
+different from [`pageshow`](https://html.spec.whatwg.org/#event-pageshow) as in the newly
+initialized document `pageshow` is only fired once the document is fully loaded.
+
+1. Add an API that allows continuing a view transition, using the current state as the old state.
```

Comment:
This will be a completely new feature for same-doc transitions also (see the thread here : https://chat.google.com/room/AAAAaLJ4XZQ/HrGp3_YjvAw). Not sure if it makes sense to mention here since we're talking about what's needed in the script API for parity with same-doc?

Reply:
The idea is that if we have that feature, we might not need additional MPA-specific APIs.

Reply:
I reworded this a bit. How does it look now @khushalsagar?